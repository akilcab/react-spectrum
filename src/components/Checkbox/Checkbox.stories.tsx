import { Story, Meta } from "@storybook/react/types-6-0";
import { Checkbox } from "./Checkbox";
import { CheckboxProps } from "@material-ui/core";

export default {
  title: "Checkbox",
  component: Checkbox,
  description: `A Checkbox.`,
  argTypes: {

  },
} as Meta;

//👇 We create a “template” of how args map to rendering
const Template: Story<CheckboxProps> = (args) => (
  <Checkbox {...args}/>
);

//👇 Each story then reuses that template
export const Default = Template.bind({});
Default.args = {
  size: "small",
  checked: true,
  name: "terms",
};
