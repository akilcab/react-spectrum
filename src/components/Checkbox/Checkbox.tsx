import React from "react";
import { Checkbox as CheckBox, CheckboxProps } from "@material-ui/core";
import { createStyles, makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      color: "#13938f",
      "&$checked": {
        color: "#13938f",
      },
    },
    checked: {},
  })
);

export const Checkbox: React.FC<CheckboxProps> = (props) => {
  const classes = useStyles(props);

  return <CheckBox className={classes.root} color="default" {...props} />;
};
