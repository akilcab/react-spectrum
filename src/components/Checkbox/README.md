# Description of component

`CheckBox` can be used to turn an option on or off.

If you have multiple options appearing in a list, you can preserve space by using checkboxes instead of on/off switches.
If you have a single option, avoid using a checkbox and use an on/off switch instead.

#### Example :

![](assets/desc.png)

## How to import

```node
import { CheckBox } from "app/shared/components/CheckBox";
```

## How to use

```node
<CheckBox
  size="small"
  checked={true}
  onChange={handleChange}
  name="terms"
/>
```

## Props

All others Checkbox props

## CSS properties we can override

You can override all CSS properties of the CheckboxProps type of Material UI.
