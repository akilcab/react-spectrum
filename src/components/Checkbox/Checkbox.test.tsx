import renderer from "react-test-renderer";
import { Checkbox } from "./Checkbox";

const handleChange = jest.fn();

describe("<Checkbox />", () => {
  test("rend correctement", () => {
    const component = renderer
      .create(
        <Checkbox
          size="small"
          checked={true}
          onChange={handleChange}
          name="terms"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
