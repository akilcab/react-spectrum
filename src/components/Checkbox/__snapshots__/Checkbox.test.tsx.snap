// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> rend correctement 1`] = `
<span
  aria-disabled={false}
  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root makeStyles-root-1 PrivateSwitchBase-checked-4 Mui-checked"
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <span
    className="MuiIconButton-label"
  >
    <input
      checked={true}
      className="PrivateSwitchBase-input-6"
      data-indeterminate={false}
      name="terms"
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </span>
</span>
`;
